File: Solution/1310.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        vector<int> prefixSum = {arr[0]};
        for (int i = 1; i < arr.size(); ++i) prefixSum.push_back(prefixSum.back() ^ arr[i]);

        vector<int> result;
        for (const auto& query : queries) {
            int left = query[0], right = query[1];
            if (left == 0) result.push_back(prefixSum[right]);
            else result.push_back(prefixSum[right] ^ prefixSum[left - 1]);
        }

        return result;
    }
};

File: Solution/1310.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        prefix_xor = [0]
        for num in arr:
            prefix_xor.append(prefix_xor[-1] ^ num)
        
        out = []
        for query in queries:
            s, e = query
            out.append(prefix_xor[e+1] ^ prefix_xor[s])
        return out